File: Algorithm/168. Excel Sheet Column Title.py
# -*- coding: utf-8 -*-
# @Author: LC
# @Date:   2016-07-19 13:45:42
# @Last modified by:   WuLC
# @Last Modified time: 2016-07-19 13:53:37
# @Email: liangchaowu5@gmail.com


class Solution(object):
    def convertToTitle(self, n):
        """
        :type n: int
        :rtype: str
        """
        result = []
        while n:
            tmp = n % 26
            if tmp == 0: 
                result.append('Z')
            else:
                result.append(chr(tmp+64))
            # the above five lines can be replaced by the following line
            # result.append(chr((n-1) % 26+65))
            n = (n-1)/26
        result.reverse()
        return ''.join(result)
        

File: Algorithm/Java/168. Excel Sheet Column Title.java
/*
 * Created on Sat Nov 04 2017 20:21:42
 * Author: WuLC
 * EMail: liangchaowu5@gmail.com
 */

// a little trick, minus 1 from n for each iteration
class Solution 
{
    public String convertToTitle(int n) 
    {
        StringBuilder result = new StringBuilder();
        while(n != 0)
        {
            n -= 1;
            result.append((char)(65 + n%26));
            n /= 26;
        }
        return result.reverse().toString();
    }
}